geode-core/src/distributedTest/java/org/apache/geode/internal/cache/PartitionedRegionClearWithConcurrentOperationsDUnitTest.java

```diff
@@ -0,0 +1,525 @@
+    Region<String, String> region = cacheRule.getCache().getRegion(REGION_NAME);
+    region.forEach(regionSnapshot::put);
+
+    return regionSnapshot;
+  }
+
+  /**
+   * Asserts that the region data is consistent on requested VMs.
+   */
+  private void assertRegionDataConsistency(List<VM> vms) {
+    // Get first snapshot to compare with the rest.
+    final Map<String, String> vm0Snapshot =
+        vms.get(0).invoke(this::waitForSilenceAndGetRegionSnapshot);
+
+    vms.forEach(vm -> vm.invoke(() -> {
+      final Map<String, String> thisSnapshot = waitForSilenceAndGetRegionSnapshot();
+      assertThat(thisSnapshot).isEqualTo(vm0Snapshot);
+      thisSnapshot.forEach((key, value) -> assertThat(value).isEqualTo("Value_" + key));
+    }));
+  }
+
+  /**
+   * Populates the region and verifies the data on the selected VMs.
+   */
+  private void populateRegion(VM feeder, int entryCount, List<VM> vms) {
+    feeder.invoke(() -> {
+      Region<String, String> region = cacheRule.getCache().getRegion(REGION_NAME);
+      IntStream.range(0, entryCount).forEach(i -> region.put(String.valueOf(i), "Value_" + i));
+    });
+
+    assertRegionData(entryCount, vms);
+  }
+
+  /**
+   * Continuously execute get operations on the PartitionedRegion for the given durationInSeconds.
+   */
+  private void executeGets(final int numEntries, final long durationInSeconds) {
+    Cache cache = cacheRule.getCache();
+    Region<String, String> region = cache.getRegion(REGION_NAME);
+    Instant finishTime = Instant.now().plusSeconds(durationInSeconds);
+
+    while (Instant.now().isBefore(finishTime)) {
+      // Region might have been cleared in between, that's why we check for null.
+      IntStream.range(0, numEntries).forEach(i -> {
+        Optional<String> nullableValue = Optional.ofNullable(region.get(String.valueOf(i)));
+        nullableValue.ifPresent(value -> assertThat(value).isEqualTo("Value_" + i));
+      });
+    }
+  }
+
+  /**
+   * Continuously execute put operations on the PartitionedRegion for the given durationInSeconds.
+   */
+  private void executePuts(final int numEntries, final long durationInSeconds) {
+    Cache cache = cacheRule.getCache();
+    Region<String, String> region = cache.getRegion(REGION_NAME);
+    Instant finishTime = Instant.now().plusSeconds(durationInSeconds);
+
+    while (Instant.now().isBefore(finishTime)) {
+      IntStream.range(0, numEntries).forEach(i -> region.put(String.valueOf(i), "Value_" + i));
+    }
+  }
+
+  /**
+   * Continuously execute clear operations on the PartitionedRegion for the given durationInSeconds.
+   */
+  private void executeClears(final long durationInSeconds, final long waitTimeInSeconds)
+      throws InterruptedException {
+    Cache cache = cacheRule.getCache();
+    Region<String, String> region = cache.getRegion(REGION_NAME);
+    Instant finishTime = Instant.now().plusSeconds(durationInSeconds);
+
+    while (Instant.now().isBefore(finishTime)) {
+      region.clear();
```

Comment:
you need to measure how long a clear() took. If it's more than 5 seconds, something must be wrong.

Reply:
I'm not following you on this one... why only 5 seconds?, and why should the test verify how long the operation actually took?. The method basically executes the `clear` operation for a specific period of time to be able to interleave this operation with other concurrent ones in the cache. Measuring how long the operation took should be part of another ticket (a performance one, probably).

Reply:
5 seconds is rvv domination timeout. If clear operation took more than 5 seconds, something must be wrong. No need to measure other operations' time.

Reply:
@gesterzhou : I still don't understand what these tests have to do with the fact that a `clear` operation should take less than 5 seconds. The ticket is to purely verify that `clear` can be executed while other cache operations are being executed at the same time on the same region, shouldn't we create an extra ticket to test the scenario you're talking about?.